### PR TITLE
SimdAsHWIntrinsic improvements and cleanup

### PR DESCRIPTION
**Pull request by tannergooding**
This does some general cleanup to the `simdashwintrinsic` code by sharing code paths where possible and using the `gtNewSimd*Node` helpers where they exist.

As part of that, this resolves an issue with the operand order used for a `gtNewSimdWithElementNode` call. I additionally attempted to finishe the cleanup to use `fgMakeMultiUse` rather than `impCloneExpr`, however it hit various asserts and needs more followup (I plan on doing this in a separate PR: https://github.com/dotnet/runtime/pull/80242).

### CODE REVIEW

**Comment by tannergooding** [comment]
The few regressions are cases where we are spilling an operand (`impSpillSideEffect`) where we weren't before, effectively a bug fix.

**Comment by vargaz** [comment]
The mono changes look ok.

**Review by fanyang-mono — APPROVED** [review]

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 1]
```diff
@@ -746,41 +756,12 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                 case NI_Vector3_get_Zero:
```
Comment (EgorBo) [inline src/coreclr/jit/simdashwintrinsic.cpp:756]: Does it handle Vector2 like previously?

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 2]
```diff
@@ -1209,280 +1185,403 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
...
+                case NI_Vector3_op_Division:
+                case NI_Vector4_op_Division:
+                case NI_VectorT128_Divide:
+                case NI_VectorT128_op_Division:
+#if defined(TARGET_XARCH)
+                case NI_VectorT256_Divide:
+                case NI_VectorT256_op_Division:
+#endif // TARGET_XARCH
+                {
+                    return gtNewSimdBinOpNode(GT_DIV, retType, op1, op2, simdBaseJitType, simdSize,
```
Comment (EgorBo) [inline src/coreclr/jit/simdashwintrinsic.cpp:1261]: I assume eventually we might need to implement the magic division optimization if op2 is CNS_VEC? 🙂 (or pdivp is fast enough as is?)

**Review by EgorBo — APPROVED** [review]

**Comment by EgorBo** [comment]
Nice [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=125651&view=ms.vss-build-web.run-extensions-tab)

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/simdashwintrinsic.cpp:756]: Yep, it's just one line outside the diff view.

Vector2/3/4 don't have a get_Count property

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 2 above)
Comment (tannergooding) [reply src/coreclr/jit/simdashwintrinsic.cpp:1261]: Right, there are some optimization opportunities here that we can more easily enable/centralize in the future with this change.

**Comment by tannergooding** [comment]
Merged slightly early (only mono llvmfullaot legs left). CI had already been passing except for two `WasmBuildTest` legs. I only merged in latest dotnet/main to ensure that failure was unrelated

**Comment by EgorBo** [comment]
Improvements on x64:

- https://github.com/dotnet/perf-autofiling-issues/issues/11466

**Pull request merged**